### PR TITLE
Replace k8s.gcr.io with registry.k8s.io

### PR DESCRIPTION
**Pull request by swastik959**
**replaced k8s.gro.io with registry.k8s.io**

- if any discrepancy is there feel free to point out 
- #28665

### CODE REVIEW

**Comment by reetasingh** [comment]
@swastik959 there are more references to "k8s.gcr.io"  in the repo. I suggest you can try `grep -rnw . -e "k8s.gcr.io"` to find all the references

**Comment by swastik959** [comment]
> @swastik959 there are more references to "k8s.gcr.io" in the repo. I suggest you can try `grep -rnw . -e "k8s.gcr.io"` to find all the references
@reetasingh thanks I have tried to change a few file please have a look

**Comment by bart0sh** [comment]
@swastik959 There are still some refs left:
```
$ git grep k8s.gcr.io
config/jobs/kubernetes/sig-k8s-infra/releng/artifact-promotion-presubmits.yaml:    run_if_changed: 'k8s.gcr.io/((images/.*/images\.yaml)|(manifests/.*/promoter-manifest\.yaml))'
jobs/e2e_node/containerd/config-systemd.toml:# Enable registry.k8s.io as the primary mirror for k8s.gcr.io
jobs/e2e_node/containerd/config-systemd.toml:  endpoint = ["https://registry.k8s.io", "https://k8s.gcr.io",]
jobs/e2e_node/containerd/config.toml:# Enable registry.k8s.io as the primary mirror for k8s.gcr.io
jobs/e2e_node/containerd/config.toml:  endpoint = ["https://registry.k8s.io", "https://k8s.gcr.io",]
```
/ok-to-test

File: jobs/e2e_node/containerd/config-systemd.toml [hunk 1]
```diff
@@ -34,7 +34,7 @@ oom_score = -999
 
 # Enable registry.k8s.io as the primary mirror for k8s.gcr.io
 # See: https://github.com/kubernetes/k8s.io/issues/3411
-[plugins."io.containerd.grpc.v1.cri".registry.mirrors."k8s.gcr.io"]
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry.k8s.io"]
```
Comment (SergeyKanzhelev) [inline jobs/e2e_node/containerd/config-systemd.toml:37]: please keep the original This is redirect that we want to preserve

File: jobs/e2e_node/containerd/config.toml [hunk 2]
```diff
@@ -28,5 +28,5 @@ oom_score = -999
 
 # Enable registry.k8s.io as the primary mirror for k8s.gcr.io
 # See: https://github.com/kubernetes/k8s.io/issues/3411
-[plugins."io.containerd.grpc.v1.cri".registry.mirrors."k8s.gcr.io"]
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry.k8s.io"]
```
Comment (SergeyKanzhelev) [inline jobs/e2e_node/containerd/config.toml:31]: please keep the original. This is by design - configured redirect (mirror)

File: label_sync/labels.md [hunk 3]
```diff
@@ -383,7 +383,7 @@ larger set of contributors to apply/remove them.
 | <a id="area/infra/aws" href="#area/infra/aws">`area/infra/aws`</a> | Issues or PRs related to Kubernetes AWS infrastructure| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
 | <a id="area/infra/monitoring" href="#area/infra/monitoring">`area/infra/monitoring`</a> | REMOVING. This will be deleted after 2021-08-04 00:00:00 +0000 UTC <br><br> Issues or PRs related to monitoring of the Kubernetes project infrastructure| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
 | <a id="area/infra/reliability" href="#area/infra/reliability">`area/infra/reliability`</a> | REMOVING. This will be deleted after 2021-08-04 00:00:00 +0000 UTC <br><br> Issues or PR related to reliability of the Kubernetes project infrastructure| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
-| <a id="area/k8s.gcr.io" href="#area/k8s.gcr.io">`area/k8s.gcr.io`</a> | Code in k8s.gcr.io/| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
+| <a id="area/registry.k8s.io" href="#area/registry.k8s.io">`area/registry.k8s.io`</a> | Code in registry.k8s.io/| anyone |  [label](https://git.k8s.io/test-infra/prow/plugins/label) |
```
Comment (SergeyKanzhelev) [inline label_sync/labels.md:386]: need to change the label first: https://github.com/kubernetes/k8s.io/labels/area%2Fk8s.gcr.io

Maybe @BenTheElder has permissions?

File: label_sync/labels.yaml [hunk 4]
```diff
@@ -1022,8 +1022,8 @@ repos:
         prowPlugin: label
         addedBy: anyone
       - color: 0052cc
-        description: Code in k8s.gcr.io/
-        name: area/k8s.gcr.io
+        description: Code in registry.k8s.io/
+        name: area/registry.k8s.io
```
Comment (SergeyKanzhelev) [inline label_sync/labels.yaml:1026]: will this be a rename or issues will drop the labels: https://github.com/kubernetes/k8s.io/labels/area%2Fk8s.gcr.io ?

**Comment by swastik959** [comment]
> @swastik959 There are still some refs left:
> ```
> $ git grep k8s.gcr.io
> config/jobs/kubernetes/sig-k8s-infra/releng/artifact-promotion-presubmits.yaml:    run_if_changed: 'k8s.gcr.io/((images/.*/images\.yaml)|(manifests/.*/promoter-manifest\.yaml))'
> jobs/e2e_node/containerd/config-systemd.toml:# Enable registry.k8s.io as the primary mirror for k8s.gcr.io
> jobs/e2e_node/containerd/config-systemd.toml:  endpoint = ["https://registry.k8s.io", "https://k8s.gcr.io",]
> jobs/e2e_node/containerd/config.toml:# Enable registry.k8s.io as the primary mirror for k8s.gcr.io
> jobs/e2e_node/containerd/config.toml:  endpoint = ["https://registry.k8s.io", "https://k8s.gcr.io",]
> ```
> /ok-to-test
> 

Do I need to change enable and the endpoint also??

File: config/jobs/kubernetes/sig-k8s-infra/trusted/releng/releng-trusted.yaml [hunk 5]
```diff
@@ -30,7 +30,7 @@ postsubmits:
     decorate: true
     decoration_config:
       timeout: 4h
-    run_if_changed: 'k8s.gcr.io/((images/.*/images\.yaml)|(manifests/.*/promoter-manifest\.yaml))'
+    run_if_changed: 'registry.k8s.io/((images/.*/images\.yaml)|(manifests/.*/promoter-manifest\.yaml))'
```
Comment (upodroid) [inline config/jobs/kubernetes/sig-k8s-infra/trusted/releng/releng-trusted.yaml:33]: Can you revert all the changes to this file please? This will break the image promotion jobs right now. I will resolve this as part of the image registry freeze in March.

File: config/jobs/kubernetes/sig-k8s-infra/releng/artifact-promotion-presubmits.yaml [hunk 6]
```diff
@@ -8,7 +8,7 @@ presubmits:
       testgrid-dashboards: sig-k8s-infra-k8sio
     decorate: true
     skip_report: false
-    run_if_changed: 'k8s.gcr.io/((images/.*/images\.yaml)|(manifests/.*/promoter-manifest\.yaml))'
```
Comment (upodroid) [inline config/jobs/kubernetes/sig-k8s-infra/releng/artifact-promotion-presubmits.yaml:11]: Can you revert all the changes to this file please? This will break the image promotion jobs right now. I will resolve this as part of the image registry freeze in March.

File: label_sync/labels.yaml (hunk 4 above)
Comment (BenTheElder) [reply label_sync/labels.yaml:1026]: I don't think we should do that. We should probably add a new label. cc @kubernetes/release-engineering @kubernetes/sig-k8s-infra

**Comment by Sajiyah-Salat** [comment]
why this issue is still open?

**Comment by BenTheElder** [comment]
@Sajiyah-Salat why wouldn't the PR still be open?

There were unaddressed review comments so it wasn't ready for merge yet.
It had only been a few days since those comments were made though.

File: config/jobs/image-pushing/README.md [hunk 7]
```diff
@@ -160,7 +160,7 @@ If you need the `.git` directory to be uploaded to your build environment:
 - add an empty `.gcloudignore` file to your repository. This will override the
   [default values][gcloudignore].
 
-[gcr instructions]: https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/README.md
+[gcr instructions]: https://github.com/kubernetes/k8s.io/blob/main/registry.k8s.io/README.md
```
Comment (BenTheElder) [inline config/jobs/image-pushing/README.md:163]: The instructions are still at https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/README.md

Can we please double check that changes are valid instead of mass substituting?

For all references to using/pulling an image, registry.k8s.io contains all images from k8s.gcr.io

For other contexts, they may not be interchangeable.

File: config/jobs/kubernetes/kops/build_jobs.py [hunk 8]
```diff
@@ -444,7 +444,7 @@ def generate_grid():
 #############################
 def generate_misc():
     results = [
-        # A one-off scenario testing the k8s.gcr.io mirror
+        # A one-off scenario testing the registry.k8s.io mirror
```
Comment (BenTheElder) [inline config/jobs/kubernetes/kops/build_jobs.py:447]: registry.k8s.io is the mirror of k8s.gcr.io referenced here

File: prow/cmd/phony/examples/review_approve_submitted.json [hunk 9]
```diff
@@ -155,7 +155,7 @@
...
...pull-requests.md#marking-unfinished-pull-requests\r\n-->\r\n\r\n**What this PR does / why we need it**:\r\nIn our e2e test suite we have use an image \"registry.k8s.io/e2e-test-images:k8s-aggregator-sample-apiserver:1.7v2\". We need a way to build a fresh image that can we can use instead of that one. Especially we need one that has a multi-arch fat manifest so e2e tests can be run across multiple architectures.\r\n\r\nThis is especially important since we are in the process of promoting the test in question to the conformance suite - https://github.com/kubernetes/kubernetes/pull/63947\r\n\r\n**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:\r\nFixes #\r\n\r\n**Special notes for your reviewer**:\r\n/cc @mkumatag \r\n/cc @ixdy \r\n/cc @luxas \r\n\r\n**Release note**:\r\n<!--  Write your release note:\r\n1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string \"action required\".\r\n2. If no release note is required, just write \"NONE\".\r\n-->\r\n```release-note\r\nNONE\r\n```\r\n",
```
Comment (marquiz) [inline prow/cmd/phony/examples/review_approve_submitted.json:158]: nit: why is this changed? just trailing whitespace added(?)

File: jobs/e2e_node/containerd/config-systemd.toml [hunk 10]
```diff
@@ -34,7 +34,7 @@ oom_score = -999
 
 # Enable registry.k8s.io as the primary mirror for k8s.gcr.io
 # See: https://github.com/kubernetes/k8s.io/issues/3411
-[plugins."io.containerd.grpc.v1.cri".registry.mirrors."k8s.gcr.io"]
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors."k8s.gro.io"]
```
Comment (marquiz) [inline jobs/e2e_node/containerd/config-systemd.toml:37]: why rename gcr -> gro?

**Comment by bart0sh** [comment]
@swastik959 Please fix CI test failures and address remaining review comments, thanks.

**Comment by BenTheElder** [comment]
Thanks for the PR.

Since this seems to be tricker than initially expected and it's been some time, I did a full pass through and filed a smaller PR with the necessary changes in https://github.com/kubernetes/test-infra/pull/28985

/close
